src/VirtualClient/VirtualClient.Actions/Furmark/FurmarkExecutor.cs

```diff
@@ -0,0 +1,257 @@
+            }
+        }
+
+        /// <summary>
+        /// Defines the name of the package associated with the component.
+        /// </summary>
+        public string PsExecPackageName
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(FurmarkExecutor.PsExecPackageName));
+            }
+        }
+
+        /// <summary>
+        /// Defines the session id of the psexec process.
+        /// </summary>
+        public int SessionId
+        {
+            get
+            {
+                return this.Parameters.GetValue<int>(nameof(this.SessionId), 1);
+            }
+        }
+
+        /// <summary>
+        /// Initializes the environment for execution of the FURMARK workload.
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            if (this.Platform != PlatformID.Win32NT)
+            {
+                throw new NotSupportedException($"'{this.Platform}' is not currently supported");
+            }
+
+            IPackageManager packageManagerWin = this.Dependencies.GetService<IPackageManager>();
+           
+            DependencyPath workloadPackage = await packageManagerWin.GetPlatformSpecificPackageAsync(this.PackageName, this.Platform, this.CpuArchitecture, cancellationToken)
+                .ConfigureAwait(false);
+
+            this.packageDirectory = workloadPackage.Path;
+
+            DependencyPath psExecPackage = await packageManagerWin.GetPlatformSpecificPackageAsync(this.PsExecPackageName, this.Platform, this.CpuArchitecture, cancellationToken)
+                .ConfigureAwait(false);
+
+            this.ExecutableLocation = this.PlatformSpecifics.Combine(this.packageDirectory, "Geeks3D", "Benchmarks", "FurMark", "Furmark");
+            this.ResultsFilePath = this.PlatformSpecifics.Combine(this.packageDirectory, "FurMark-Scores.txt");
+            this.XMLFilePath = this.PlatformSpecifics.Combine(this.packageDirectory, "Geeks3D", "Benchmarks", "FurMark", "furmark-gpu-monitoring.xml");
+            this.PSexecExecutablePath = this.PlatformSpecifics.Combine(psExecPackage.Path, "psexec.exe");
+        }
+
+        /// <summary>
+        /// Executes the FURMARK workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            using (BackgroundOperations profiling = BackgroundOperations.BeginProfiling(this, cancellationToken))
+            {
+                await this.ExecuteWorkloadAsync(telemetryContext, cancellationToken)
+                    .ConfigureAwait(false);
+            }
+        }
+
+        private async Task ExecuteWorkloadAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            if (File.Exists(this.ResultsFilePath))
+            {
+                File.Delete(this.ResultsFilePath);
```

Comment:
Please check how we are doing it in other executors, please use function from systemmanagement, u will not need to check if file exists or not, that function is already checking